### DISCUSSION TITLE
Recommended recipe for including mathjax fails pre-commit hooks

### DISCUSSION THREAD

**Discussion by gpap-gpap** [mkdocs/mkdocs#3831] [Maintenance]
I have followed the [recommended recipe](https://squidfunk.github.io/mkdocs-material/reference/math/) for including mathjax in my docs, however when I do so my pre commit hooks fail the standard check:
```
standard.................................................................Failed
- hook id: standard
- exit code: 1

standard: Use JavaScript Standard Style (https://standardjs.com)
'document$' is not defined. (no-undef)
'MathJax' is not defined. (no-undef)
```

I am a total javascript noob, so not sure if this has to do with the way I have configured my pre-commit or with the way the recipe is suggested (or if this is even a genuine issue worthy of a ticket). Could someone suggest a fix or next step?

**Comment by pawamoy** [comment]
Tagging @squidfunk and @kamilkrzyskow since the linked recipe comes from Material for MkDocs.

**Reply by kamilkrzyskow** [reply]
I'm not a JS Wizard, but the pre-commit hook seems to be checking if the `document$` and `MathJax` are defined and accessible in the `extra.js` file. The issue is that `document$` comes from the Material bundle.js and is declared at runtime, and the `MathJax` variable is declared at the start via the `window.MathJax` and is probably extended later with the `tex-mml-chtml.js`. 

`extra.js` has access to those values, because it runs after the execution of `bundle.js` and after the `document$` was declared.
I would add some sort of "ignore check" comments for `extra.js` code that accesses variables from outside the `extra.js` file itself, ~~but I don't know how to do that~~ 😉 

EDIT:
A quick google leads to this:
- https://eslint.org/docs/latest/rules/no-undef
- `/*global document$*/`

This should help with the `document$`, but I'm not sure about the `MathJax`, as it's declared with `window.MathJax` 🤔

**Reply by squidfunk** [reply]
`document$` and `window.document$` are the same, since in-browser `window` == `globalThis`, so `/* global MathJax */` might work as well. But yes, both are global variables, defined by Material for MkDocs and MathJax.

**Reply by gpap-gpap** [reply]
Thanks a lot for the replies and detective work
```
/* global document$ */
/* global MathJax */
```
did indeed solve it!

**Comment by gpap-gpap** [comment]
Adding
```
/* global document$ */
/* global MathJax */
```
at the top of `mathjax.js` resolves the issue
